client/src/components/Bookings/Booking.tsx

```diff
@@ -0,0 +1,52 @@
+import Box from '@material-ui/core/Box';
+import Typography from '@material-ui/core/Typography';
+import Card from '@material-ui/core/Card';
+import CardContent from '@material-ui/core/CardContent';
+import useStyles from './useStyles';
+import AvatarDisplay from '../AvatarDisplay/AvatarDisplay';
+import SettingsIcon from '@mui/icons-material/Settings';
+
+interface Props {
+  bookingType: string;
+  timing: string;
+  clientName: string;
+  status: string;
+}
+
+const Booking = ({ bookingType, timing, clientName, status }: Props): JSX.Element => {
+  const classes = useStyles();
+  return (
+    <Card className={bookingType === `next` ? classes.nextBooking : classes.bookings}>
+      <Box display="flex" justifyContent="space-between">
+        <CardContent>
+          {bookingType === 'next' && (
+            <Typography className={classes.title} variant="h6" gutterBottom>
+              Your Next Booking:
+            </Typography>
+          )}
+          <Typography className={classes.pos} variant="h3">
+            {timing}
+          </Typography>
+          <Box display="flex" alignItems="center">
+            <AvatarDisplay />
+            <Typography className={classes.clientName} variant="h3" gutterBottom>
+              {clientName}
+            </Typography>
+          </Box>
+        </CardContent>
+        {bookingType === 'next' ? (
+          <SettingsIcon fontSize="small" />
+        ) : (
+          <Box display="flex" flexDirection="column" alignItems="end" justifyContent="space-around">
+            <SettingsIcon fontSize="small" />
+            <Typography className={classes.status} color="textSecondary" variant="h6" gutterBottom>
+              {status}
+            </Typography>
+          </Box>
+        )}
+      </Box>
+    </Card>
+  );
+};
```

Comment:
After the discussion, I wanted to elaborate on this. Let's take this UI breakdown as an example (just one way to interpret): 

<img width="379" alt="ui-breakdown-example" src="https://user-images.githubusercontent.com/70963670/144119715-b11fd7ac-2913-4a27-8446-b8c8a3dc8c8c.png">

![Blue](https://via.placeholder.com/15/3f48cc/000000?text=+) = `Card/Paper`,
![Red](https://via.placeholder.com/15/ed1c24/000000?text=+) = `Title`,
![Green](https://via.placeholder.com/15/22b14c/000000?text=+) = `Subtitle`,
![Yellow](https://via.placeholder.com/15/ffc90e/000000?text=+) = `Content`,
![Purple](https://via.placeholder.com/15/c8bfe7/000000?text=+) = `Settings/Action`,

Since both cards have the same structure, it makes sense that we could use the same component for them. We can recieve a prop for each of the sections listed (for example, `title`, `subtitle` as strings and `children` can be for content...). We can make the title/subtitle optional, we can style the component so it has "responsive" width (e.g. `width: 100%` could work well) and we can optionally receive props for things like [`elevation`](https://v4.mui.com/components/paper/#demo-z725m2mdhre), [`variants`](https://v4.mui.com/components/paper/#variants) etc.

More than the specific implementation, what's important is that this version of the component can be used for many cases (e.g. listing users, reviews... virtually anything). This is mainly because the structure of it is *decoupled* from its content - you'll notice this version doesn't take any props related to a "Booking"... Instead it's adaptable! 🎉 

That's the idea I wanted to emphasize here - reusable abstractions often come from removing as much context from the implementation as possible. 

It's not always easy to identify this at a first glance though, it's a skill that develops over time so take it at your own pace - it's okay to come back to this idea another time - but I would be doing you a disservice if I didn't mention it 👍🏼

Reply:
Thanks @ugur-akin for your valuable suggestions